index.js

```diff
@@ -375,6 +375,144 @@ BabelClient.prototype.createAnnotation = function(token, data, options, callback
+    }
+    if (!data._id) {
+        throw new Error('Missing data: _id');
+    }
+    if (!data.hasBody) {
+        throw new Error('Missing data: hasBody');
+    }
+    if (!data.hasBody.format) {
+        throw new Error('Missing data: hasBody.format');
+    }
+    if (!data.hasBody.type) {
+        throw new Error('Missing data: hasBody.type');
+    }
+    if (!data.annotatedBy) {
+        throw new Error('Missing data: annotatedBy');
+    }
+    if (!data.hasTarget) {
+        throw new Error('Missing data: hasTarget');
+    }
+
+    // validate the hasTarget property
+    var targets = [];
+    if (_.isArray(data.hasTarget)) {
+        targets = data.hasTarget;
+        if (targets.length === 0) {
+            throw new Error('Missing data: hasTarget cannot be empty array');
+        }
+    } else {
+        targets.push(data.hasTarget);
+    }
+    _.map(targets,function(target) {
+        if (!_.has(target, 'uri')) {
+            throw new Error('Missing data: hasTarget.uri is required');
+        }
+        for (var prop in target) {
+            if (!(prop === 'uri' || prop === 'fragment' || prop === 'asReferencedBy' )) {
+                throw new Error('Invalid data: hasTarget has unrecognised property \'' + prop + '\'');
+            }
+        }
+    });
+
+    var requestOptions = {
+        method: 'PUT',
+        body: data,
+        json: true,
+        url: this.config.babel_host + ':' + this.config.babel_port + '/annotations/' + data._id,
+        headers: {
+            'Accept': 'application/json',
+            'Authorization': 'Bearer ' + token,
+            'Host': this.config.babel_hostname
+        }
+    };
+
+    this.debug(JSON.stringify(requestOptions));
+
+    request.put(requestOptions, function(err, response, body){
+        if (err) {
+            callback(err);
+        } else {
+            if (body.message && body.errors) {
+                var babelError = new Error(body.message);
+                babelError.http_code = response.statusCode || 404;
+                callback(babelError);
+            } else {
+                callback(null, body);
+            }
+        }
+    });
+};
+
+/**
+ * Delete an annotation by its id
+ *
+ * @param {string} token Persona token
+ * @param {object} annotationId Annotation id to delete
+ * @param callback
+ */
+BabelClient.prototype.deleteAnnotation = function(token, annotationId, callback){
+    if (!token) {
+        throw new Error('Missing Persona token');
+    }
+    if (!annotationId) {
+        throw new Error('Missing annotationId');
+    }
+
+    var requestOptions = {
+        method: 'DELETE',
+        json: true,
+        url: this.config.babel_host + ':' + this.config.babel_port + '/annotations/' + annotationId,
+        headers: {
+            'Accept': 'application/json',
+            'Authorization': 'Bearer ' + token,
+            'Host': this.config.babel_hostname
+        }
+    };
+
+    this.debug(JSON.stringify(requestOptions));
```

Comment:
surely the point here is that this is a debug message and won't be seen in prod?